src/applications/check-in/day-of/pages/CheckIn/index.jsx

```diff
@@ -48,7 +96,13 @@ const CheckIn = props => {
     [dispatch, context, goToErrorPage, token],
   );
 
-  if (!appointment) {
+  if (isDayOfDemographicsFlagsEnabled) {
+    if (!appointment || !demographicsFlagsSent) {
+      return (
+        <va-loading-indicator message="Loading your appointments for today" />
+      );
+    }
+  } else if (!appointment) {
     return (
       <va-loading-indicator message="Loading your appointments for today" />
     );
```

Comment:
```suggestion
  if (!appointment || (isDayOfDemographicsFlagsEnabled && !demographicsFlagsSent)) {
    return (
      <va-loading-indicator message="Loading your appointments for today" />
    );
```

I had to 🤔 on this a bit but I think this is equivalent?

Reply:
Looks much better, thanks @acrollet ! I ended up removing this block entirely though since we're no longer making the demographics call on the CheckIn page with the new implementation